calico/felix/devices.py

```diff
@@ -33,6 +33,23 @@
 _log = logging.getLogger(__name__)
 
 
+def check_kernel_config():
+    """
+    Checks the kernel configuration for problems that would break our
+    security guarantees, for example.
+
+    :raises BadKernelConfig if a problem is detected.
+    """
+    ps_name = "/proc/sys/net/ipv4/conf/all/rp_filter"
+    rp_filter = int(_read_proc_sys(ps_name).strip())
+    if rp_filter > 1:
+        _log.critical("Kernel's RPF check is set to 'loose'.  This would "
+                      "allow endpoints to spoof their IP address.  Calico "
+                      "requires net.ipv4.conf.all.rp_filter to be set to "
+                      "0 or 1.")
```

Comment:
The kernel uses the maximum value between the "all" and "<interface>" values but the ordering of the values is

* 0 == no checking
* 1 == strict
* 2 == loose

We want strict checking so we need the all value to be set to 0 or 1, then we can set 1 on the interface and get the right result.  If it's set to 2 then that overrides our per-interface setting.